lib/nosedrum/component_handler/ets.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Nosedrum.ComponentHandler.ETS do
+  @moduledoc """
+  ETS-based implementation of a `Nosedrum.ComponentHandler`.
+
+  When a pid is registered as a handler for components it will be monitored 
+  and automatically removed once the process exists. Registered module handlers 
+  on the other hand are never cleared. You can access the named ETS-table manually
+  to remove any keys if necessary.
+
+  ## Options
+  * :name - name used for registering the process under and also the name of the ETS-table. 
+  Must be a atom, because it will be passed to `:ets.new/2` as name.
+  """
+  use GenServer
+
+  @behaviour Nosedrum.ComponentHandler
+
+  alias Nosedrum.Storage
+
+  @impl Nosedrum.ComponentHandler
+  def register_components(server \\ __MODULE__, component_ids, module) do
+    GenServer.call(server, {:register_components, component_ids, module})
+  end
+
+  @impl Nosedrum.ComponentHandler
+  def handle_component_interaction(
+        server \\ __MODULE__,
+        %Nostrum.Struct.Interaction{} = interaction
+      ) do
+    component_id = interaction.data.custom_id
+
+    case :ets.match(server, {component_id, :"$1"}) do
+      [[pid]] when is_pid(pid) ->
+        send(pid, {:message_component_interaction, interaction})
+        :ok
+
+      [[module]] when is_atom(module) ->
+        with response <- module.message_component_interaction(interaction),
+             {:ok} <- Storage.respond(interaction, response),
+             {_defer_type, callback_tuple} <- Keyword.get(response, :type) do
+          Storage.followup(interaction, callback_tuple)
+        end
+
+      _ ->
```

Comment:
Should we explicitly match on `[[]]` here to make sure we don't have garbage in the ETS table?

Reply:
I don't use ETS often, so I'm not sure if that's necessary. Under which circumstances would this match return `[[]]`?

Reply:
Sorry, I think I meant `[]`. I'm asking for it because the other two explicitly match on the type and if by _tiny_ chance we ever have something else inserted e.g. by a bug we can catch it here.

Reply:
Ah, that makes more sense. Good idea!